flang/runtime/command.cpp

```diff
@@ -222,6 +253,22 @@ std::int32_t RTNAME(GetCommand)(const Descriptor *value,
   return stat;
 }
 
+std::int32_t RTNAME(GetLog)(const Descriptor *value, const Descriptor *errmsg) {
+  FillWithSpaces(*value);
+
+  const char *arg = getlogin();
```

Comment:
Can we use `getlogin_r` instead of `getlogin()`? getlogin isn't thread safe.
e.g:
```
std::array<char, LOGIN_NAME_MAX> arg;
int err = getlogin_r(arg.data(), arg.size());
if (err != 0) {
    // error
}
// Shrink the string to the right length
std::string str(arg);
return CopyToDescriptor(*value, str.data(), str.size(), errmsg);
```
(I haven't tested this it's just an example)

Then we don't need the function local static on Windows either as we can emulate the getlogin_r interface there.

Reply:
I have changed to `getlogin_r` in commit: https://github.com/llvm/llvm-project/pull/70917/commits/e0df6ab9683a070e2c7e94325a26f68d369736f4